xapian-core/weight/tfidfweight.cc

```diff
@@ -141,18 +141,18 @@ double
 TfIdfWeight::get_idfn(Xapian::doccount termfreq, char c) const
 {
     double N = 1.0;
-    if (c != 'n') N = get_collection_size();
+    if (c != 'n' || c != 'f') N = get_collection_size();
```

Comment:
Yes, this is always true but I realized it shouldn't be! This has to be `&&` because we don't want to fetch N in both the cases not just one of them. I'll make the change. Thanks.